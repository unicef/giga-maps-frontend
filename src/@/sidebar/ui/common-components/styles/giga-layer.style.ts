import styled from "styled-components";

import { Scroll } from "~/@/scroll";

export const SingleGigaLayerItem = styled.div<{ $isActive?: boolean; $disabled?: boolean }>`
background: ${(prop) => prop.$isActive ? prop.theme.titleBlue : prop.theme.gigaButtonBack};
padding:0.25rem;
cursor: pointer;
width: inherit;
height: inherit;
display: flex;
flex-direction: column;
justify-content: space-between;

.single-giga-layer-icon {
display: flex;
    align-items: center;
    justify-content: space-between;
.layer-icon{
  width: 1rem;
  height: 1rem;
  fill:${props => props.$isActive ? props.theme.white : props.theme.titleDesc};
}
.information-icon{

  button{
    border: none;
    background: inherit;
  }

  svg{
    width: 0.625rem;
  height: 0.625rem;
  fill:#C8C8C8;
}
}
}

.single-giga-layer-name {
margin-top:0.5rem;
p{
color: ${props => props.$isActive ? props.theme.white : props.theme.titleDesc};
font-size: 0.75rem;
font-weight: 400;
line-height: 1rem;
letter-spacing: -0.01rem;
display: -webkit-box;
-webkit-box-orient: vertical;
-webkit-line-clamp: 2;
overflow: hidden;
text-overflow: ellipsis;
word-break: break-word;
}
}
`
export const SidePanelLayerWrapper = styled.div<{ $wrap?: boolean; }>`
display: flex;
  width: 100%;
  height: 100%;
  flex-wrap: ${(prop) => prop.$wrap ? "wrap" : "nowrap"} ;
`

export const SidePanelLayerGirdColumns = styled.div<{ margin: string }>`
display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
  width: 4.9rem;
  height: 4.12rem;
  margin:${(prop) => prop.margin};
  min-width: 4.9rem;
  position: relative;
  .disabled-box {
    background: rgba(223,223,223,0.4);
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
    cursor: not-allowed;
    z-index: 1;
  }
  @media only screen and (min-width: 1584px) {
    width: 5.05rem;
    min-width: 5.05rem;
  }
`

export const GigaLayerText = styled.p`
color: ${props => props.theme.text};
font-size: 0.75rem;
font-weight: 400;
line-height: 1.125rem;
letter-spacing: 0.01rem;
`

export const GigaPopUpScroll = styled(Scroll)`
max-height: 11rem; 
`