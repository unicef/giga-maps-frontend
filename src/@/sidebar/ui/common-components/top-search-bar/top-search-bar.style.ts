import { Button, IconButton, Search } from '@carbon/react';
import styled, { css } from 'styled-components';

export const SearchContainer = styled(Search) <{ $isMobile: boolean }>`
/* .cds--search-input:focus{
  outline:none;
} */

input{
padding: 0 2.5rem 0 0.5rem !important;
border-bottom: 1px solid ${props => props.theme.main};
background: ${props => props.theme.main};
color: ${props => props.theme.text};
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.125rem;
}
input::placeholder{
  color: #6F6F6F;
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.125rem;
}
.cds--search-magnifier{
  
  svg{
    display: ${props => props.value ? 'none' : 'inline-block'};
    fill:${props => props.theme.text};
    left: 9.125rem
  }
}
.cds--search-close{
  svg{
    fill:${props => props.theme.text};
  }
  :hover{
    background: transparent;
    border-block-end:1px solid transparent
  }

}
.cds--search-close::before{
  background-color: transparent;
  block-size: 0;
}
.cds--search-close:hover, .cds--search-button:hover {
  background-color:  transparent;
}
.cds--search-close
button{
  :hover{
    background: transparent;
  }
}
${props => props.$isMobile && css`
  /* .cds--search { */
    position: fixed;
    top: 0;
    left: 0;
  /* } */
  .cds--search-close--hidden {
    opacity: 1;
    visibility: visible;
  }
`}

`

export const SearchWrapper = styled.div`
display:flex;
/* flex-direction: row-reverse; */
align-items:center;
height:3.1rem;

.cds--btn--primary:hover{
    background-color: ${props => props.theme.main} !important; 
  }

  .cds--tooltip-content {
  background: ${props => props.theme.text} !important;
  color: ${props => props.theme.main}!important;
}
  .cds--popover-caret{
    background: ${props => props.theme.text};
  }
  
.search-icon{
  margin-right: 0.5rem;
  height: 1rem;
  width: 1rem;
  fill:#fff;
}
svg{
    fill:${props => props.theme.text} !important ;
  }
@media (max-width:768px){
  position: fixed;
  /* width: calc(100% - 2rem); */
  width: auto;
  top: 0rem;
  z-index: 2;
  right: 0;
  flex-direction: row-reverse;
}

`

export const BackButton = styled.div`
    height: 100%;
    display: flex;
    align-items: center;
    padding-right: 0.5rem;
    padding-left:0.5rem;
    `
export const CountrySearchIcon = styled(Button)`

/* background:rgb(0, 0, 0); */
background: ${props => props.theme.main};
padding-right: 0.5rem;
padding-left: 0.688rem;
.label-size {
  font-size: 0.75rem;
}
.cds--btn--primary:hover{
    background-color: ${props => props.theme.main} !important; 
  }
  .cds--btn--primary:active {
    background-color: ${props => props.theme.main} !important;
}

svg{
  margin-right: 0.20rem;
  fill:${props => props.theme.text};
}

@media (max-width:768px){
  background: ${props => props.theme.main};

  .cds--btn--primary:hover{
    background-color: ${props => props.theme.main} !important; 
  }
  .cds--btn--primary:active {
    background-color: ${props => props.theme.main} !important;
}

}
`