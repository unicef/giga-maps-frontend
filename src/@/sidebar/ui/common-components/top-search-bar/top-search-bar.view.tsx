import { ChevronDown, Earth, Search } from '@carbon/icons-react'
import { useStore } from 'effector-react';
import { useState } from 'react';

import { $isMobile } from '~/core/media-query';
import { getVoid } from '~/lib/effector-kit';
import { getInputValue } from '~/lib/event-reducers';

import { $searchInput, $showCountries, changeIsSearchFocused, changeSearchText, clearSearchText, onShowCountriesAdminList, } from './top-search-bar.model';
import { CountrySearchIcon, SearchContainer, SearchWrapper } from './top-search-bar.style';
import { useTranslation } from 'react-i18next';


const onChange = changeSearchText.prepend(getInputValue);
const onClear = clearSearchText.prepend(getVoid);

const TopSearchBar = () => {
  const searchText = useStore($searchInput);
  const showCountries = useStore($showCountries)
  const isMobile = useStore($isMobile)
  const [mobileSearch, setMobileSearch] = useState(false)
  const { t } = useTranslation();

  const onBlurSearch = () => {
    setTimeout(() => {
      changeIsSearchFocused(false)
    }, 300)
  }
  return (
    <SearchWrapper className="top-search-bar">
      <CountrySearchIcon
        align={'bottom-left'}
        label='Country List'
        className='main-search-list'
        onClick={() => {
          onShowCountriesAdminList(!showCountries)
        }}
        size="lg"
        kind="primary"
      >
        <Earth />
        <ChevronDown />
      </CountrySearchIcon>
      {isMobile && !mobileSearch ? (
        <Search
          className='search-icon'
          onClick={() => { setMobileSearch(true); }}
        />
      ) : null}
      {(isMobile && mobileSearch || !isMobile) && <SearchContainer
        size="lg"
        placeholder={t("search-country-region-school-id")}
        labelText="Search"
        closeButtonLabelText="Clear search input"
        id="main-search-bar"
        autoFocus={isMobile}
        onChange={onChange}
        onFocus={() => changeIsSearchFocused(true)}
        onBlur={onBlurSearch}
        value={searchText}
        className={"sidebar-searchbox"}
        $isMobile={isMobile}
        onClear={() => {
          onClear();
          isMobile && setMobileSearch(false);
          changeIsSearchFocused(false);
        }}
      />}
    </SearchWrapper >
  );
};

export default TopSearchBar;
