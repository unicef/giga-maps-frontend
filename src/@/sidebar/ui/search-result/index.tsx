
import { useStore } from 'effector-react';
import FooterTourContact from '../common-components/footer-tour-contact.view';
import { $hasSearchInput, $isSearchFocused, $showCountries } from '../common-components/top-search-bar/top-search-bar.model';
import { SearchCountryList } from './search-country-list';
import SearchSchoolPanel from './search-country-list/search-school-panel-view';
import { SearchListWrapper, SearchResultScroll, SearchResultWrapper } from './styles/search-result-style';
import SearchResultList from './views/search-result.list.view';
import { useTranslation } from 'react-i18next';

export default function SearchResult() {
  const isSearchFocus = useStore($isSearchFocused);
  const showCountries = useStore($showCountries);
  const hasSearchInput = useStore($hasSearchInput);
  const { t } = useTranslation();
  return (
    <>
      {showCountries && <SearchListWrapper>
        <SearchResultScroll className="search-container">
          <SearchCountryList />
        </SearchResultScroll>
        < FooterTourContact message={t("not-the-results-you-expected")} />
        < SearchSchoolPanel />
      </SearchListWrapper>}
      {isSearchFocus && hasSearchInput &&
        <SearchResultWrapper>
          <SearchResultScroll>
            <SearchResultList />
          </SearchResultScroll>
          <FooterTourContact message={t("not-the-results-you-expected")} />
        </SearchResultWrapper>
      }
    </>
  );
};
