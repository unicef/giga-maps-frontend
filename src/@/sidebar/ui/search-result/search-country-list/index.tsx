
import { useStore } from 'effector-react';
import { useEffect } from 'react';

import { fetchCountriesWithDistrictFx } from '../container/search-result.fx';
import { $searchCountryList } from '../container/search-result.model';
import { SearchTopHead } from '../styles/search-result-style';
import { SearchCountry } from './search-country-view';
import SearchResultLoading from './search-result-loading.view';
import { useTranslation } from 'react-i18next';

export const SearchCountryList = () => {
  const searchCountryList = useStore($searchCountryList);
  const isLoading = useStore(fetchCountriesWithDistrictFx.pending);
  const { t } = useTranslation();
  useEffect(() => {
    if (!searchCountryList?.length) {
      void fetchCountriesWithDistrictFx(null);
    }
  }, [searchCountryList]);

  if (isLoading) return <SearchResultLoading />;
  return (<>
    {/* <SearchHistory /> */}
    <SearchTopHead as="p">{t('select-country-region-school-s')}</SearchTopHead>
    {searchCountryList.map((countryData) => <SearchCountry key={countryData.country_id} countryData={countryData} />)}
  </>);
};
