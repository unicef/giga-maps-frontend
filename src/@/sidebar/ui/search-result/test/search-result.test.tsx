import { describe, expect, test, } from '@jest/globals';
import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

import { changeIsSearchFocused, changeSearchText, onShowCountriesAdminList } from '../../common-components/top-search-bar/top-search-bar.model';
import SearchResult from '..';
import { getSearchResultsFx } from '../container/search-result.fx';
import SearchResultList from '../views/search-result.list.view';
import { SearchResultWrapper } from '../styles/search-result-style';
import { testWrapper } from '~/tests/jest-wrapper';
import "~/core/i18n/instance"

describe('SearchResultList', () => {
  beforeEach(() => {
    fetchMock.mockResponse((req: Request) => {
      if (req.url.includes('api/locations/search-countries/')) {
        return Promise.resolve(JSON.stringify([{
          admin1_count: 1,
          country_code: "AI",
          country_id: 12,
          country_name: "India",
          data: {
            "A": {
              "admin2_name": "East End",
              "admin2_id": 4929,
              "admin2_description": "district",
              "admin2_description_ui_label": "district",
              "admin2_code": "AIA001002",
              "school_count": 1,
              data: {
                "B": {
                  admin2Code: "A2"
                }
              }
            }
          }
        }]))
      } else if (req.url.includes('/locations/gsearch/')) {
        return Promise.resolve(JSON.stringify({
          results: [{
            "country_name": "Argentina",
            "giga_id_school": "33ad9ec6-a5e5-3930-982c-3e3297260a6a",
            "id": 1231556,
            "admin1_name": "CÃ³rdoba",
            "country_id": 143,
            "admin2_name": "Capital",
            "external_id": "",
            "name": "ESCUELA REPUBLICA DE LA INDIA",
            "country_code": "AR",
            "@search.score": 1.0,
            "@search.highlights": null
          }]
        }))
      }
    })
  })
  test('renders SearchResult and take a snapshop', () => {
    onShowCountriesAdminList(true);
    const { asFragment } = render(<SearchResult />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders SearchResult', async () => {
    onShowCountriesAdminList(true);
    render(<SearchResult />);
    expect(screen.queryByText('Not the results you expected?')).toBeInTheDocument();
  });

  test('renders SearchResult click outside', async () => {
    onShowCountriesAdminList(true);
    render(<SearchResult />);
    await userEvent.click(document.body);
    expect(screen.queryByText('Select region or school')).not.toBeInTheDocument();
  });

  test('renders SearchResult and onShowCountriesAdminList is false', async () => {
    render(<SearchResult />);
    await onShowCountriesAdminList(false);
    expect(screen.queryByText('Select region or school')).not.toBeInTheDocument();
  });

  test('renders SearchResultList with loading state', async () => {
    changeIsSearchFocused(true);
    const spy = jest.spyOn(getSearchResultsFx.pending, 'getState');
    spy.mockReturnValue(true);
    await render(<SearchResultList />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
    spy.mockRestore();
  });


  test('renders SearchResultList with India Country', async () => {
    changeSearchText('India')
    await getSearchResultsFx({ query: 'India' })
    await render(<SearchResultList />);
    expect(screen.getByText('ESCUELA REPUBLICA DE LA')).toBeInTheDocument();
  });

  it('should render component', () => {
    const { asFragment } = render(testWrapper(<SearchResultWrapper />));
    expect(asFragment).toMatchSnapshot();
  });
})