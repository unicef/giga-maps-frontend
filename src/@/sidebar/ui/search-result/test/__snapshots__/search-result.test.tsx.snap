// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultList renders SearchResult and take a snapshop 1`] = `
<DocumentFragment>
  .c9 {
  padding: 1rem 0;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 0.75rem;
  flex-direction: column;
}

.c11 {
  display: flex;
  align-items: center;
}

.c11 a:hover {
  text-decoration: underline!important;
}

.c10 {
  font-size: 0.75rem;
}

.c12 {
  font-size: 0.75rem!important;
  font-style: normal;
  font-weight: 400;
  cursor: pointer;
}

.c0 {
  position: absolute;
  width: 100%;
  height: calc(100% - 6rem);
  z-index: 5;
  top: 6rem;
}

.c1 {
  height: calc(100% - 4rem);
}

.c2 {
  display: flex;
  align-items: center;
  height: 2rem;
  padding: 1rem 1rem;
  font-size: 0.75rem;
}

.c4 {
  padding: 1rem;
  justify-content: space-between;
  align-items: center;
  display: flex;
  width: calc(100% - 1rem);
  margin-left: .5rem;
  border-bottom: 0.0625rem solid;
}

.c4 .type-name {
  display: block;
}

.c4 .highlight {
  color: var(--primary-blue);
}

.c4 .light {
  color: var(--text-text-tertiary);
}

.c4 .sidebar-history-remove >svg {
  fill: !important;
}

.c5 {
  font-size: 0.875rem;
  position: relative;
  max-width: 8rem;
}

.c5 a:hover {
  text-decoration: underline;
}

.c7 {
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.c8 {
  pointer-events: none;
}

.c3 {
  text-align: left;
  padding: 0!important;
}

.c6 {
  text-align: left;
  margin-block-end: 0rem;
  padding: 0!important;
}

@media (max-width:768px) {
  .c0 {
    position: fixed;
    top: 3.1rem;
    height: calc(100% - 3rem);
    z-index: 11;
  }
}

<div
    class="c0"
  >
    <div
      class="scrollbar-container c1 search-container ps"
    >
      <div
        class="ps__rail-x"
        style="left: 0px; top: 0px;"
      >
        <div
          class="ps__thumb-x"
          style="left: 0px; width: 0px;"
          tabindex="0"
        />
      </div>
      <div
        class="ps__rail-y"
        style="top: 0px; left: 0px;"
      >
        <div
          class="ps__thumb-y"
          style="top: 0px; height: 0px;"
          tabindex="0"
        />
      </div>
      <span
        class="c2"
      >
        <p
          class="cds--skeleton__text c3"
          style="width: 80%;"
        />
      </span>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="c4"
      >
        <h4
          class="c5"
          style="width: 40%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
        </h4>
        <div
          class="c7"
          style="width: 30%;"
        >
          <p
            class="cds--skeleton__text c6"
            style="width: 100%;"
          />
          <svg
            aria-hidden="true"
            class="0 c8"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <p
        class="c10"
      >
        Not the results you expected?
      </p>
      <div
        class="c11"
      >
        <a
          class="cds--link c12"
        >
          Take the tour
        </a>
        <span>
           or 
        </span>
        <a
          class="cds--link c12"
          href="/about#live-map-get-in-touch"
          rel="noopener"
          target="_blank"
        >
          Contact us
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SearchResultList should render component 1`] = `[Function]`;
